src/components/input_validator.vue

```diff
@@ -0,0 +1,60 @@
+<template>
+  <slot :props="{ onInput, errorMessages, onBlur }"></slot>
+</template>
+
+<script lang="ts" setup>
+import { onMounted, ref } from 'vue'
+import { watch } from 'vue'
+
+export type Rule = (value?: any) => string
```

Comment:
```suggestion
export type Rule = (value?: any) => string | null | undefined | true
```

Don't strict return to string
make it string (non empty one) when it's invalid other wise accept any falsy value